tests/cache/tests.py

```diff
@@ -1009,6 +1009,26 @@ def test_get_or_set_racing(self):
             cache_add.return_value = False
             self.assertEqual(cache.get_or_set('key', 'default'), 'default')
 
+    def test_validate_key_called(self):
+        tests = [
+            ('add', ['a', 1]),
+            ('get', ['a']),
+            ('set', ['a', 1]),
+            ('incr', ['a']),
+            ('decr', ['a']),
+            ('touch', ['a']),
+            ('delete', ['a']),
+            ('get_many', [['a', 'b']]),
+            ('set_many', [{'a': 1, 'b': 2}]),
+            ('delete_many', [{'a': 1, 'b': 2}]),
+        ]
+        with mock.patch.object(cache, 'validate_key', autospec=True) as m:
+            for method, args in tests:
+                with self.subTest(method=method):
+                    m.reset_mock()
+                    getattr(cache, method)(*args)
+                    m.assert_called()
```

Comment:
Can we add tests that will catch `InvalidCacheKey` or `CacheKeyWarning` instead of mocking `validate_key()`?

We should also move tests for methods which work without this patch to a separate commit, e.g. `set()`, `add()`, `incr()`, `decr()`, or `set_many()`.

Reply:
I figured there was already a test for the behaviour of `validate_key()` so we just need to check it was being called, but yes we can change it.